src/DOM/HTML/Event/DataTransfer.purs

```diff
@@ -61,3 +66,27 @@ setData
   -> DataTransfer
   -> Eff (dom :: DOM | eff) Unit
 setData (MediaType format) dat dt = setDataImpl format dat dt
+
+foreign import dropEffectImpl :: forall eff. DataTransfer -> Eff (dom :: DOM | eff) String
+
+data DropEffect = None | Copy | Move | Link
+
+--| Gets the data transfer object's drop effect.
+dropEffect :: forall eff. DataTransfer -> Eff (dom :: DOM | eff) DropEffect
+dropEffect dt = do
+  de <- dropEffectImpl dt
+  pure $ unsafePartialBecause "No other values are possible https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect" case de of
+    "copy" -> Copy
+    "move" -> Move
+    "link" -> Link
+    "none" -> None
```

Comment:
`unsafePartialBecause` is problematic (https://github.com/purescript/purescript-partial/issues/7), could you use `crashWith` in an `_` case, and then `unsafePartial` the `case` instead please?